recipes/csv-parser/all/conanfile.py

```diff
@@ -0,0 +1,45 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class CsvParserConan(ConanFile):
+    name = "csv-parser"
+    description = "Vince's CSV Parser with simple and intuitive syntax"
+    topics = ("conan", "csv-parser", "csv", "rfc 4180", "parser", "generator")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/vincentlaucsb/csv-parser"
+    license = "MIT"
+    settings = "os", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        tools.check_min_cppstd(self, 11)
+
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(self.settings.compiler.version)
+        if compiler == "gcc" and compiler_version < "7":
+            raise ConanInvalidConfiguration("gcc version < 7 not supported")
+
```

Comment:
To summarize, there does not exist a `-std=c++XX` option for gcc 6 that makes it compile?
If it would compile when you pass `-std=c++17`, then you should add that test as well.

Reply:
I tested again with -std=c++14 and -std=c++17, with no luck. This appears to be a [bug](https://stackoverflow.com/questions/36489369/constexpr-non-static-member-function-with-non-constexpr-constructor-gcc-clang-d) that was fixed in gcc 7.2. 
The author of the library seems to have introduced [some code](https://github.com/vincentlaucsb/csv-parser/blob/f65b6a72ece010f0bee7cfb3ef91b0e07f14eabc/single_include/csv.hpp#L4747) to fix this (excerpt here):
```c++
#if defined __GNUC__ && !defined __clang__
    #if (__GNUC__ >= 7 &&__GNUC_MINOR__ >= 2) || (__GNUC__ >= 8)
        #define CONSTEXPR constexpr
    #endif
    #else
        #ifdef CSV_HAS_CXX17
        #define CONSTEXPR constexpr
    #endif
#endif`
```
However the error occurs on a line using a *CONSTEXPR_14* define, so the fix does not affect this particular error on gcc6.
I would probably leave the gcc >= 7 restriction in.

Reply:
I just now tested with clang5, and this requires setting *-std=c++11* for test_package to work.
What is the preferred way of dealing with this? Do i just
```python
if self.settings.compiler == clang and self.settings.compiler.version == 5:
    self.cpp_info.cxxflags += ["-std=c++11"]
```
?

Reply:
No don't handle it like that, because then you force users to c++11, even if they want c++14/17/... or the compiler default.
We currently don't do anything c++ standard related in `package_info`.

What we do is check in `validate` whether the compiler supports it and check `self.settings.compiler.cppstd` if it is set.

Reply:
And I'm totally fine with your current handling. I was just asking.